src/services/db/GitFileSystemService.ts

```diff
@@ -350,11 +374,13 @@ export default class GitFileSystemService {
               .commit(commitMessage),
             (error) => {
               if (error instanceof GitError) {
-                return new GitFileSystemError(error.message)
+                return new GitFileSystemNeedRollbackError(error.message)
```

Comment:
do we need a new orElse branch to call `rollback`?

Reply:
Hmm I don't have a preference for this. My original intention was to let the respective functions (create, update, delete) handle the rollback, as it is possible that something bad happens during the `fs.promises` step. So it is possible to rollback here in the commit function, but the original calling functions would have also performed a rollback themselves, so I avoided adding it here.

Reply:
... we are re-implementing transactions here arent we :( 

+ 1, we should have a guaranteed rollback incase fs.promises throws t